src/symbolize/gimli.rs

```diff
@@ -340,17 +341,20 @@ thread_local! {
     // leverage the structures built when constructing `addr2line::Context`s to
     // get nice speedups. If we didn't have this cache, that amortization would
     // never happen, and symbolicating backtraces would be ssssllllooooowwww.
-    static MAPPINGS_CACHE: RefCell<Vec<(PathBuf, Mapping)>>
-        = RefCell::new(Vec::with_capacity(MAPPINGS_CACHE_SIZE));
+    static mut MAPPINGS_CACHE: Option<Cache> = None;
+
+    unsafe { f(MAPPINGS_CACHE.get_or_insert_with(|| Vec::with_capacity(MAPPINGS_CACHE_SIZE))) }
+}
+
+pub fn clear_symbol_cache() {
```

Comment:
I think this is currently not threadsafe since this doesn't acquire any locks, but this function should be `unsafe` (surfaced from `with_cache`) and then the wrapper in `symbolize/mod.rs` should grab the lock. Additionally the function in `symbolize/mod.rs` I think should only be enabled for now with the `std` feature enabled.

Reply:
Hmm yeah, I didn't realize that now that was all non-threadsafe. I've never used `unsafe` before and I'm too used to Rust telling me when something I'm doing is dumb :)
I guess it means it works!

Anyway, I'll make the changes you suggested